### ISSUE TITLE
Which tslint.json is used?

### ISSUE THREAD

**Issue by Sebazzz** [rich-newman/typescript-analyzer#25]
I attempted to use the extension in [one project](https://github.com/Sebazzz/financial-app/tree/b81ba0f7b1f6a5099c6f04640e0e2bac3c2fdb86) of mine.

tslint is installed locally in the only project of the solution. In the project is one tslint.json config file. However, I'm getting messages about quotemarks not matching though the settings in my tslint are configured otherwise. I also attempted to put the tslint.json in the solution folder after reading some issues here and extend the project tslint, but without any effect.

**Comment by rich-newman** [comment]
The analyzer looks in the folder of the first file it's given to lint for a tslint.json, and uses that if there is one.  If there's not it looks in the parent folder, then the parent's parent and so on until it runs out of parents.  If it still hasn't found a tslint.json it uses the default, which is usually c:\Users\{Username}.

Whilst this isn't very sophisticated it works for most scenarios.  In short, if you put a tslint.json in the solution folder and check there aren't any tslint.jsons anywhere else in your folders, and assuming a fairly normal project layout, then that should work.  Let me know if it's not.

**Comment by Sebazzz** [comment]
I think if figured out what might be going wrong. tslint is also invoked on my webpack javascript configuration files. The false positives only happen there. Checking the regular files causes no problems as far as I can see now.

**Comment by jduhls** [comment]
Please consider not putting tslint.json directly in home dir root. I have put in a request for VS to create a home folder so it does not interfere with other development:

https://visualstudio.uservoice.com/forums/121579-visual-studio-ide/suggestions/33544837-create-vs-folder-in-user-home-dir-do-not-allow

**Comment by rich-newman** [comment]
Yes, I'm going to be weaselly here and say that I did just inherit this from Mads' original extension.  It's not too hard to make it configurable, although I expect that isn't the problem.  My guess is you install the extension and it blows up an existing dev environment because it's stuck a tslint.json in c:/Users/{username}.  Is that right?  If not, what circumstances are we trying to fix?  I don't think tslint knows anything about that folder.

**Comment by jduhls** [comment]
That is correct.  tslint as a node_module in any project crawls up the project tree and also at some point checks the users's home folder until it finds a tslint.json. I would argue strongly that the tslint.json in the user's home folder should ONLY be generated BY CHOICE by the user and is even expected to be. If this extension needs a default it should be in some place that belongs specifically to the extension or at least to VS. I don't think the root of the users's home folder should belong to the extension.
